File: scala/sdktest/CreateResultEntityTest.scala
// Generated basic-flow test for the create_result entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped CreateResultTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefindecryptxp2pesdk.core.{Helpers, BluefinDecryptxP2peSDK}
import voxgig.bluefindecryptxp2pesdk.utility.struct.Struct

object CreateResultEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("create_result.instance") {
      val testsdk = BluefinDecryptxP2peSDK.testSDK()
      val ent = testsdk.createResult(null)
      rep.check("create_result.instance", ent != null, "expected non-null create_result entity")
    }

    rep.scope("create_result.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/create_result/CreateResultTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinDecryptxP2peSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("create_result01", "CREATE_RESULT01")
      idmap.put("create_result02", "CREATE_RESULT02")
      idmap.put("create_result03", "CREATE_RESULT03")
      idmap.put("device01", "DEVICE01")
      idmap.put("device02", "DEVICE02")
      idmap.put("device03", "DEVICE03")
      idmap.put("device_type01", "DEVICE_TYPE01")
      idmap.put("serial_number01", "SERIAL_NUMBER01")
      val now = System.currentTimeMillis()

      // CREATE
      val createResultRef01Ent = client.createResult(null)
      var createResultRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.create_result"), "create_result_ref01"))
      createResultRef01Data.put("device_type", idmap.get("device_type01"))
      createResultRef01Data.put("serial_number", idmap.get("serial_number01"))
      val createResultRef01DataResult = createResultRef01Ent.create(createResultRef01Data, null)
      createResultRef01Data = Helpers.toMapAny(createResultRef01DataResult match { case e: SdkEntity => e.data(); case o => o })
      rep.check("create_result.create.map", createResultRef01Data != null, "expected create result to be a map")
    }
  }
}

File: elixir/test/create_result_entity_test.exs
# CreateResult entity test (offline, mock transport)

defmodule BluefinDecryptxP2pe.CreateResultEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinDecryptxP2pe.Helpers, as: H
  alias BluefinDecryptxP2pe.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/create_result/CreateResultTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinDecryptxP2pe.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.create_result"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinDecryptxP2pe.test()
    ent = BluefinDecryptxP2pe.create_result(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = BluefinDecryptxP2pe.test(S.jm(["entity", S.jm(["create_result", S.jm([])])]))
    ent = BluefinDecryptxP2pe.create_result(sdk)
    created = BluefinDecryptxP2pe.Entity.CreateResult.create(ent, S.jm(["name", "test-create"]))
    made = BluefinDecryptxP2pe.EntityBase.data_get(created)
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end
